test/functional/lua/buffer_updates_spec.lua

```diff
@@ -305,6 +305,19 @@ describe('lua buffer event callbacks: on_lines', function()
     n.assert_alive()
   end)
 
+  it('no invalid lnum error for closed memline in on_detach #31251', function()
+    exec_lua([[
+      vim.api.nvim_buf_set_lines(0, 0, -1, false, { '' })
+      local bufname = 'buf2'
+      local buf = vim.api.nvim_create_buf(false, true)
+      vim.api.nvim_buf_set_name(buf, bufname)
+      vim.bo[buf].bufhidden = 'wipe'
+      vim.cmd('vertical diffsplit '..bufname)
+      vim.api.nvim_buf_attach(0, false, { on_detach = function() vim.cmd("redraw") end})
```

Comment:
to ensure the test is "alive" should we assert that `on_detach` was actually called, e.g.

```
      eq(false, n.fn.exists('did_redraw'))

      ...

      vim.api.nvim_buf_attach(0, false, { on_detach = function()
        vim.g.did_redraw = true
        vim.cmd("redraw")
      end})

      ...

      eq(true, n.fn.exists('did_redraw'))
```

Reply:
Sure? I took the fact that the test fails on `master` as being sufficient.